### PR TITLE
Fixes for issues #165 and #167

### LINKED ISSUE #165: Shorten description in ?cdm_disambiguate_cols
https://github.com/krlmlr/dm/pull/158/files#r347016441

### LINKED ISSUE #167: Simplify ?cdm_enum_fk_candidates
https://github.com/krlmlr/dm/pull/158/files#r347023515

### PR DESCRIPTION
**Pull request by bbecane**
Please check.

Closes #165.

Closes #167.

### CODE REVIEW

File: R/disambiguate.R [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
-#' Key columns will not be renamed because only one column should remain when two tables that are
-#' linked by a key relation are joined.
+#' This function ensures that all columns in `dm` have unique names.
+#'
+#' The function first checks if there are any column names that are not unique.
+#'
+#' If there are, those columns will be assigned new, unique, names by prefixing their existing name
+#' with the name of their table and a separator.
+#'
+#' Columns that act as primary or foreign keys will not be renamed because only the foreign key column will remain
```
Comment (krlmlr) [inline R/disambiguate.R]: Can you please split this sentence?

File: R/foreign-keys.R [hunk 2]
```diff
@@ -164,27 +163,30 @@ cdm_rm_fk <- function(dm, table, column, ref_table) {
...
+#' that key as the `by` argument of join() to match it to each column of `table`.
+#'
+#' On most data sources, this operation implicitly also tests for type compatibility
+#' between the columns of the two tables.
+#'
+#' The outcome of the join operation determines the value of the `why` column in the result:
+#'
+#' - an empty value for a column of `table` that is a suitable foreign key candidate
+#' - the count and percentage of missing matches for a column that is not suitable
+#' - a triggered error message that may include the types of mismatched columns
```
Comment (krlmlr) [inline R/foreign-keys.R]: Attempting to join incompatible columns triggers an error, the error message is captured in this case. How to phrase this?

File: R/foreign-keys.R [hunk 3]
```diff
@@ -164,27 +163,30 @@ cdm_rm_fk <- function(dm, table, column, ref_table) {
...
 #'
 #' @return A table that lists which columns of `table` would be suitable candidates for
```
Comment (krlmlr) [inline R/foreign-keys.R]: The table lists all columns. Maybe something like:

A table that lists, for all columns, whether they are suitable ... . For columns that are not suitable, the reason is explained in the `why` column.

**Comment by bbecane** [comment]
Fixed in https://github.com/krlmlr/dm/pull/181

**Comment by krlmlr** [comment]
Thanks! I moved #181 to this pull request.

**Pull request merged**